### PR TITLE
add priority class to prow components

### PR DESCRIPTION
**Pull request by droslean**

### CODE REVIEW

**Comment by droslean** [comment]
/hold

Checking....

**Comment by stevekuznetsov** [comment]
Can we define our own priority class? Our pods are not as critical as, say, the apiserver

**Comment by droslean** [comment]
> Can we define our own priority class? Our pods are not as critical as, say, the apiserver

Yea, something like `system-prow-critical`

**Comment by droslean** [comment]
@stevekuznetsov Added a priority class dedicated to prow components only. PTAL

**Comment by petr-muller** [comment]
LGTM

File: cluster/ci/config/prow/openshift/prow-priority-class.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+apiVersion: scheduling.k8s.io/v1beta1
+description: Used for prow pods that must run in the cluster, but can be
+  moved to another node if necessary.
+kind: PriorityClass
+metadata:
+  name: system-prow-critical
+value: 1999999000
```
Comment (stevekuznetsov) [inline cluster/ci/config/prow/openshift/prow-priority-class.yaml]: How did you arrive at this?

Comment (droslean) [reply cluster/ci/config/prow/openshift/prow-priority-class.yaml]: `system-node-critical` is `2000001000`
`system-cluster-critical` is `2000000000` so I followed the patern here and put `2000000000 - 1000 = 1999999000`

Comment (stevekuznetsov) [reply cluster/ci/config/prow/openshift/prow-priority-class.yaml]: SGTM

**Comment by stevekuznetsov** [comment]
Does the config-updater have privileges to create the service class? Otherwise merging this will take down Prow :)

**Comment by droslean** [comment]
@stevekuznetsov rbacs added. But, even now, it will fail to create the `priorityclass` because of the order of the update that we do in the `Makefile`. It seems that we are not updating the `config-updater` roles in the `postsubmit-update`. That leaves us with two options. We can create the roles manually before merging this PR, or we should change the order that things are happening in the `Makefile`. Any thoughts?

**Comment by stevekuznetsov** [comment]
You could also make two PRs, right?

**Comment by stevekuznetsov** [comment]
Or `oc apply -f` the roles yourself then we can `/hold cancel` this one

**Pull request merged**